explainer.md

```diff
@@ -0,0 +1,459 @@
+});
+```
+
+### New Document
+```js
+requestAnimationFrame(() => {
+  let pendingTransition = document.documentTransition.getPendingTransition();
+  if (pendingTransition.getData().version !== 123)
+    return;
+  
+  // |offeredTransitionElements| provides a list of objects to access state
+  // saved from the old page.
+  let oldHeader = pendingTransition.offeredTransitionItems.get("header-id");
+  if (oldHeader) {
+    // Add sharedId attribute to elements animated in the new DOM.
+    document.querySelector(".header").sharedId="header-id";
+    
+    // Query the style information saved from the old page.
+    let oldHeaderStyle = oldHeader.getContainerComputedStyle();
+    
+    // The pseudo elements for each shared element are associated with the root element.
+    // The existing [pseudoElement](https://drafts.csswg.org/web-animations-1/#dom-keyframeeffect-pseudoelement) option can be used
+    // to target them with Web Animations API.
+    document.documentElement.animate(
+      [{ width: oldHeaderStyle.width,
+        height: oldHeaderStyle.height,
+        transform: oldHeaderStyle.transform }],
+      { duration: 1000,
+        pseudoElement: '::shared-container(#header-id)' });
+  }
+});
+```
+
+# Alternatives Considered
+## Heirarchical Properties
+This proposal disallows a shared element to be nested inside another shared element. The restriction avoids the need to preserve the hierarchy of the shared elements and associated properties (transform, clip, effects inherited by descendents) when creating pseudo elements. This is a consideration for future iterations of the feature.
+
+## Container/Child Split
+One consideration was to render each shared element using a replaced element directly instead of creating a container element. The motivation behind this split is to provide a stacking context to cross-fade the content of old and new shared elements. This was necessary to ensure blending identical pixels is a no-op using [plus-lighter](https://drafts.fxtf.org/compositing/#porterduffcompositingoperators_plus_lighter) blending. While same-origin transitions could work around this, it enables future extensibility for cross-origin transitions where cross-fading identical images would be common.
+
+## Natively Supporting Animating Box Decoration CSS Properties
+An alternate approach to the setup described in [Animating Box Decoration CSS Properties](#live-animatable-properties) is to support this natively in the browser by introducing a new content-element() function. This function would behave similar to the element() function except skipping the following properties when painting the element: box decorations and visual effects which generate a stacking context. The image will also be sized to the element's content-box (as opposed to the border-box used by the element() function). The motivation for supporting this natively would be to make these properties animatable instead of requiring developer side changes.
```

Comment:
```suggestion
An alternate approach to the setup described in [Animating Box Decoration CSS Properties](#live-animatable-properties) is to support this natively in the browser by introducing a new `content-element()` function. This function would behave similarl to the `element()` function except skipping the following properties when painting the element: box decorations and visual effects which generate a stacking context. The image will also be sized to the element's content-box (as opposed to the border-box used by the element() function). The motivation for supporting this natively would be to make these properties animatable instead of requiring developer side changes.
```